Tests/FluentAssertions.Specs/Xml/XElementAssertionSpecs.cs

```diff
@@ -1228,5 +1228,185 @@ public void When_asserting_element_has_a_child_element_with_an_empty_name_it_sho
+
+            // Act / Assert
+            xElement.Should().BeSingle();
+        }
+
+        [Fact]
+        public void When_asserting_xElement_is_not_a_single_child_element_it_should_fail()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                    <child />
+                  </parent>");
+
+            // ReSharper disable once PossibleNullReferenceException
+            // Root is never null
+            var xElement = document.Root.Element("child");
+
+            // Act
+            Action act = () => xElement.Should().BeSingle();
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage(
+                "Expected child to have a count of 1, but found 2.");
+        }
+
+        [Fact]
+        public void When_asserting_a_null_xElement_to_be_single_it_should_fail()
+        {
+            // Arrange
+            XElement xElement = null;
+
+            // Act
+            Action act = () => xElement.Should().BeSingle();
+
+            // Assert
+            act.Should().Throw<InvalidOperationException>().WithMessage(
+                "Cannot assert the count if the element itself is <null>.");
+        }
+
+        #endregion
+
+        #region HaveCount
+
+        [Fact]
+        public void When_asserting_document_has_a_child_element_with_a_count_of_two_and_it_does_it_should_succeed()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                    <child />
+                  </parent>");
+
+            // Act / Assert
+            document.Should().HaveElement("child").Which.Should().HaveCount(2);
+        }
+
+        [Fact]
+        public void When_asserting_document_has_a_child_element_with_a_count_of_two_and_it_does_not_it_should_fail()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                    <child />
+                    <child />
+                  </parent>");
+
+            // Act
+            Action act = () => document.Should().HaveElement("child").Which.Should().HaveCount(2);
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage(
+                "Expected child to have a count of 2, but found 3.");
+        }
+
+        [Fact]
+        public void When_asserting_xElement_has_a_count_of_two_and_it_does_it_should_succeed()
+        {
+            // Arrange
+            var document = XDocument.Parse(
+                @"<parent>
+                    <child />
+                    <child />
+                  </parent>");
+
+            // ReSharper disable once PossibleNullReferenceException
+            // Root is never null
+            var xElement = document.Root.Element("child");
+
+            // Act / Assert
+            xElement.Should().HaveCount(2);
```

Comment:
How can this happen?
The method `Element` returns ["the first (in document order) child element with the specified XName"](https://docs.microsoft.com/en-us/dotnet/api/system.xml.linq.xcontainer.element?view=net-5.0#definition).
So it IS single and cannot have a count of 2!?

Reply:
Same as in the previous comment.